plugins/woocommerce/src/Blocks/Utils/BlockTemplateUtils.php

```diff
@@ -229,6 +242,21 @@ public static function build_template_result_from_file( $template_file, $templat
 		$template->post_types     = array(); // Don't appear in any Edit Post template selector dropdown.
 		$template->area           = self::get_block_template_area( $template->slug, $template_type );
 
+		/*
+		* Run the block hooks algorithm introduced in WP 6.4 on the template content.
+		*/
+		if ( function_exists( 'inject_ignored_hooked_blocks_metadata_attributes' ) ) {
+			$before_block_visitor = '_inject_theme_attribute_in_template_part_block';
+			$after_block_visitor  = null;
+			$hooked_blocks        = get_hooked_blocks();
+			if ( ! empty( $hooked_blocks ) || has_filter( 'hooked_block_types' ) ) {
+				$before_block_visitor = make_before_block_visitor( $hooked_blocks, $template );
+				$after_block_visitor  = make_after_block_visitor( $hooked_blocks, $template );
+			}
+			$blocks            = parse_blocks( $template->content );
+			$template->content = traverse_and_serialize_blocks( $blocks, $before_block_visitor, $after_block_visitor );
```

Comment:
If you're wondering why we run the `parse_blocks` and `traverse_and_serialize_blocks` functions outside of the if statement it is because the visitor needs to look for Template Part blocks, and inject a theme attribute (set to the current theme’s name) into each of them. This predates Block Hooks; it was done a bit differently before, but essentially also required parsing and traversal.

You can find the same in WordPress core [here](https://github.com/WordPress/wordpress-develop/blob/trunk/src/wp-includes/block-template-utils.php#L560C1-L565C109), where we are mimicking this.